zip-0213.rst

```diff
@@ -0,0 +1,169 @@
+  in non-coinbase transactions MUST also be applied to coinbase transactions.
+
+- Only transparent outputs in coinbase transactions are subject to the existing
+  restrictions on spending coinbase funds. More precisely:
+
+  - The existing consensus rule requiring transactions that spend coinbase outputs to have
+    an empty ``vout``, is amended to only apply to transactions that spend transparent
+    coinbase outputs.
+
+  - The existing consensus rule requiring coinbase outputs to have 100 confirmations
+    before they may be spent (coinbase maturity), is amended to only apply to transparent
+    coinbase outputs.
+
+- Full Sapling note decryption MUST succeed using the all-zero outgoing viewing key. More
+  precisely, all Sapling outputs in coinbase transactions MUST have valid note commitments
+  when recovered using a 32-byte array of zeroes as the outgoing viewing key.
+
+
+Interaction with the Founders' Reward
+-------------------------------------
+
+This ZIP does not alter the existing Founders' Reward addresses.
+
+
+Rationale
+=========
+
+The ZIP does not require that all coinbase must be shielded immediately from activation of
+the network upgrade, so that miners and mining pools may gradually migrate from their
+existing transparent addresses to Sapling addresses. This also simplifies the consensus
+rules, because the Founders' Reward targets transparent addresses, and thus it remains
+necessary for the time being to support them. A future ZIP could require all coinbase to
+be shielded immediately.
+
+Enforcing coinbase maturity at the consensus level for Sapling outputs would incur
+significant complexity in the consensus rules, because it would require special-casing
+coinbase note commitments in the Sapling commitment tree. The standard recommendation when
+spending a note is to select an anchor 10 blocks back from the current chain tip, which
+acts as a de-facto 10-block maturity on all notes, coinbase included. This might be
+proposed as a consensus rule in future.
+
+There is another reason for shielded coinbase maturity being unnecessary: when a reorg or
+rollback occurs that would cause a shielded coinbase output to disappear, it will also
+invalidate every shielded transaction that uses an anchor descending from the tree that
+the shielded coinbase output had been appended to. That is, all economic activity would be
+rolled back in addition to the shielded coinbase output disappearing, so there is no
+reason to make shielded coinbase a special case when the same behaviour occurs in regular
+shielded notes already. In the transparent coinbase case, only direct child transactions
+of the transparent coinbase would become invalid, and thus it would be possible to end up
+in a situation where a logical child transaction (for example, a mining pool paying out
+miners) persists in the block chain after its logical parent (the mining pool receiving a
+block) disappears.
```

Comment:
How can such a child transaction "persist in the block chain"? It becomes invalid, right?

Reply:
A "logical" child isn't necessarily a cryptographic child. For example, if a mining pool receives coinbase as UTXO `Q`, and then makes a payout for the block that `Q` was mined in, using UTXO `M` from before `Q` was mined, then it is possible for a reorg to cause `Q` to disappear but still leave `M` spent.